src/lib.rs

```diff
@@ -135,6 +189,28 @@ impl CommandHandle {
     pub fn kill(&self) {
         let _ = self.kill_trigger.initiate_kill();
     }
+
+    pub fn iter(&self) -> HandleIterator {
+        HandleIterator {
+            channel: &self.channel,
+        }
+    }
+}
+
+impl Iterator for CommandHandle {
+    type Item = OutputMessage;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.channel.recv().ok()
+    }
+}
+
+impl<'a> Iterator for HandleIterator<'a> {
+    type Item = OutputMessage;
+
+    fn next(&mut self) -> Option<OutputMessage> {
+        self.channel.recv().ok()
+    }
```

Comment:
You implement Iterator twice here, and the implementation looks identical. Is one of them unnecessary?

Reply:
This provides options to the caller.
they can iterate on the handle directly, but this consumes the handle, or they can create an iterator that just borrows the handle:
`for msg in handle { ...` (this consumes the handle)
`for msg in handle.iter()` (this allows the handle to be used after the loop, like joining on the handle to get exit statuses etc...)

Reply:
That's useful! Clippy (in pedantic mode, which you could run here to get an additional improvement) has pointed me to do this in my code:

```
it is more concise to loop over references to containers instead of using explicit iteration methods
  --> src/commands/setup.rs:17:20
   |
17 |     for env_app in environment.applications.iter() {
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: to write this more concisely, try: `&environment.applications`
```

In my example, `environment.applications` is a `Vec<Application>`.

Is it possible to implement your code so that one can do this:


```rust
// similar to handle.iter()
for msg in &handle { ...
// like before, consumes handle
for msg in handle { ...
```

If possible, that would feel even nicer and cleaner.

Reply:
Yeah agreed, added